capstone/src/main/java/com/google/step/clients/RestaurantClient.java

```diff
@@ -46,6 +44,11 @@ public void putRestaurant(Restaurant restaurant, String ownerEmail) {
     restaurantInfo.setProperty("score", 2.5);
 
     datastore.put(restaurantInfo);
+    // Datastore will not set the ID until the entity is put into it, so must edit the entity
+    // to set the ID once it has been generated
+    long id = restaurantInfo.getKey().getId();
+    restaurantInfo.setProperty("restaurantKey", id);
+    datastore.put(restaurantInfo);
```

Comment:
interesting, until just now I had assumed the actual key was the restaurant ID.

How come we have a separate "restaurantKey" field inside the entity? Can we not just populate the RestaurantInfo type from the entity, such that we grab the key from the entity and populate it? It seems kind of circuitous to duplicate this data.

Reply:
Great point! I think we originally put it as a separate field because we thought we had to query by fields and couldn't query based on single IDs. I figured out how to do that, though, so the latest commit has no more restaurantKey property!